api/services/get.js

```diff
@@ -0,0 +1,152 @@
+'use strict';
+
+var self = get;
+module.exports = self;
+
+var async = require('async');
+var _ = require('underscore');
+var fs = require('fs');
+var path = require('path');
+
+var configHandler = require('../../common/configHandler.js');
+
+function get(req, res) {
+  var bag = {
+    reqQuery: req.query,
+    resBody: {},
+    initializeDefault: false,
+    component: 'services',
+    defaultService: {}
+  };
+
+  bag.who = util.format('services|%s', self.name);
+  logger.info(bag.who, 'Starting');
+
+  async.series([
+    _checkInputParams.bind(null, bag),
+    _get.bind(null, bag),
+    _getServicesJson.bind(null, bag),
+    _setDefault.bind(null, bag)
+  ],
+    function (err) {
+      logger.info(bag.who, 'Completed');
+      if (err)
+        return respondWithError(res, err);
+
+      sendJSONResponse(res, bag.resBody);
+    }
+  );
+}
+
+function _checkInputParams(bag, next) {
+  var who = bag.who + '|' + _checkInputParams.name;
+  logger.verbose(who, 'Inside');
+
+  if (bag.reqQuery.name)
+    bag.serviceQuery = bag.reqQuery.name;
+
+  return next();
+}
+
+function _get(bag, next) {
+  var who = bag.who + '|' + _get.name;
+  logger.verbose(who, 'Inside');
+
+  configHandler.get(bag.component,
+    function (err, services) {
+      if (err)
+        return next(
+          new ActErr(who, ActErr.DBOperationFailed,
+            'Failed to get ' + bag.component, err)
+        );
+
+      if (_.isEmpty(services)) {
+        logger.debug('No configuration in database for ' + bag.component);
+        bag.initializeDefault = true;
+        return next();
+      }
+
+      if (bag.serviceQuery) {
+        if (!services[bag.serviceQuery])
+          return next(
+            new ActErr(who, ActErr.DataNotFound,
+              util.format('No service \'%s\' found in systemSettings.services',
+                bag.serviceQuery), err)
+          );
+        bag.resBody = services[bag.serviceQuery];
+
+      } else {
+        bag.resBody = services;
```

Comment:
bag.resBody should be an array and it should return either a single service or all services based on query param